### ISSUE TITLE
Script will not run in home folder on OSX

### ISSUE THREAD

**Issue by looneym** [gorakhargosh/watchdog#401]
I wish to monitor my home directory and all subdirectories for changes. 

I am running the simple logger example verbatim in my home directory but it does not detect changes in either the home dir or any of the subdirs. 

Moving the script to a director inside of home makes it work.

![watchdog_issue](https://cloud.githubusercontent.com/assets/12705417/23628026/de8c8696-02aa-11e7-9eea-462768789cf9.gif)

Both files in this example are identical:

```python
import sys
import time
import logging
from watchdog.observers import Observer
from watchdog.events import LoggingEventHandler

if __name__ == "__main__":
    logging.basicConfig(level=logging.INFO,
                        format='%(asctime)s - %(message)s',
                        datefmt='%Y-%m-%d %H:%M:%S')
    path = sys.argv[1] if len(sys.argv) > 1 else '.'
    event_handler = LoggingEventHandler()
    observer = Observer()
    observer.schedule(event_handler, path, recursive=True)
    observer.start()
    try:
        while True:
            time.sleep(1)
    except KeyboardInterrupt:
        observer.stop()
    observer.join()
```

Anything I'm missing here?

**Comment by amagliul** [comment]
I've noticed the same thing just using `watchmedo log` on MacOS 10.12.3. If the path specified is the user's home folder, no events will be logged.
Works: `watchmedo log --debug-force-fsevents --patterns="*" --recursive "$HOME/Desktop/"`
Doesn't work: `watchmedo log --debug-force-fsevents --patterns="*" --recursive "$HOME/"`

**Comment by sp-1234** [comment]
Same problem here. macOS 10.12.6

**Comment by looneym** [comment]
I feel like this was probably because I had a bunch of nested subdirectories in `~` so it hit a max recursion depth. Complete and total speculation though

**Comment by sp-1234** [comment]
Nope, it's just because file-level notifications in fsevents do not work. The problem is likely on macOS side and unfixable.

**Comment by looneym** [comment]
Hmm swap it out with a polling observer? 

https://github.com/looneym/sobotka/blob/master/sobotka/file_sync_utility.py#L35

**Comment by sp-1234** [comment]
polling observer for entire home folder won't fly obviously

**Comment by looneym** [comment]
I think it's unlikely to be fixed. This project is pretty much dormant.
